### PR TITLE
Combine the session of captures, manager and processors

### PR DESCRIPTION
**Pull request by shafreeck**
This work is based on #337,  merge it only after #337.

### What problem does this PR solve? 

The captures and processors use separate sessions to keep their liveness and the owner uses a session to do election. This PR combines all these three types of sessions to one. When the session expires, the ownership, capture info and processor info are all removed by one time.

### What is changed and how it works?
* Remove role manager, it is no more useful by design.
* Share one session between captures, processors and the election.
* Support to suicide itself for captures
* Support to recover from a suicide

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Breaking backward compatibility

GanttDue: 2020-04-03

### CODE REVIEW

**Comment by amyangfei** [comment]
Plesae resolve conflicts @shafreeck

File: cdc/capture.go [hunk 1]
```diff
@@ -78,99 +77,77 @@ func NewCapture(pdEndpoints []string) (c *Capture, err error) {
...
+	return c.election.Campaign(ctx, c.info.ID)
+}
+
+// Resign lets a owner start a new election.
+func (c *Capture) Resign(ctx context.Context) error {
+	return c.election.Resign(ctx)
+}
+
+// Suicide kills the capture itself
+func (c *Capture) Suicide() error {
```
Comment (amyangfei) [inline cdc/capture.go]: this is not used now?

File: cdc/capture.go [hunk 2]
```diff
@@ -23,34 +23,33 @@ import (
...
 	"google.golang.org/grpc/backoff"
 )
 
 const (
-	ownerRunInterval  = time.Millisecond * 500
 	captureSessionTTL = 3
 )
 
+// ErrSuicide causes a panic
+var ErrSuicide = errors.New("Suicide")
```
[the remark this answers is not in the document]
Comment (shafreeck) [reply cdc/capture.go]: The ErrSuicide is only used for comparing, the stack does not matter.

File: cdc/capture.go (hunk 1 above)
Comment (shafreeck) [reply cdc/capture.go]: fixed

**Pull request merged**